include/insns/RV64I.h

```diff
@@ -144,7 +144,8 @@ namespace SST{
       }
 
       static bool addw(RevFeature *F, RevRegFile *R,RevMem *M,RevInst Inst) {
-        R->RV64[Inst.rd] = dt_u64(td_u64(R->RV64[Inst.rs1],64) + td_u64(R->RV64[Inst.rs2],64),64);
+        R->RV64[Inst.rd] = dt_u32(td_u32(R->RV64[Inst.rs1],32) + td_u32(R->RV64[Inst.rs2],32),32); //addw operates on and produces 32-bit results even on RV64I codes 
+        SEXTI(R->RV64[Inst.rd], 32);    //Sign extend the result up to 64bits
```

Comment:
addw operates explicitly on 32-bit values so the input casting was updated an then  we sign extend the result